src/stack_aggregator.py

```diff
@@ -369,11 +369,11 @@ def create_dependency_data_set(resolved, ecosystem):
 
     for pv in resolved:
         if pv.get('package') and pv.get('version'):
-            key = ecosystem + "::" + pv.get('package') + "::" + pv.get('version')
+            key = ecosystem + "|" + pv.get('package') + "|" + str(pv.get('version'))
             unique_epv_dict['direct'][key] = set()
             for trans_pv in pv.get('deps', []):
-                trans_key = ecosystem + "::" + trans_pv.get('package') + "::" + \
-                            trans_pv.get('version')
+                trans_key = ecosystem + "|" + trans_pv.get('package') + "|" + \
+                            str(trans_pv.get('version'))
```

Comment:
Version not being string will have unintended consequences at graph and gremlin query. We better need to fix at the client side.

Reply:
There shouldn't be any problem in typecasting the version to string. we have to convert it to string for concatenating and creating the payload for the gremlin.

Reply:
My worry is version not being a string. Is it for a specific package? Is it for a specific ecosystem? Is it how the client generates the versions? Does this affect only a subset of packages? We need to investigate these things. Can you please share which EPV was creating this issue?

Reply:
actually, I created my own payload with the version float instead of a string. I did not use any client to generate the payload for me.
something like this:
```json
{"package": "urllib3", "version": 1.22}
```